Source/WebKit/WebProcess/GPU/graphics/RemoteImageBufferProxy.h

```diff
@@ -118,7 +118,10 @@ class RemoteImageBufferProxy : public WebCore::ConcreteImageBuffer<BackendType>
 
     void setNeedsFlush(bool needsFlush) final
     {
+        bool hasPrepared = m_needsFlush;
         m_needsFlush = needsFlush;
+        if (!hasPrepared && m_needsFlush)
+            prepareForBackingStoreChange();
     }
```

Comment:
This function does more than what its name is saying. Also if `needsFlush` is false, it will just set the flag. It will call `prepareForBackingStoreChange()` only if `m_needsFlush` changes from false to true. I would suggest the following. Leave setNeedsFlush() unchanged and add a new function named something like `backingStoreWillChange()`. This new function will be called from `RemoteDisplayListRecorderProxy::send()`.
```
void backingStoreWillChange() final
{
    if (m_needsFlush)
        return;
    prepareForBackingStoreChange();
    m_needsFlush = true;
}
```

Reply:
Did this this way. It's rather same way as in the first patch, though.